src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResponseResource.java

```diff
@@ -0,0 +1,116 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.ZonedDateTime;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.Complaint;
+import de.tum.in.www1.artemis.domain.ComplaintResponse;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.ComplaintRepository;
+import de.tum.in.www1.artemis.repository.ComplaintResponseRepository;
+import de.tum.in.www1.artemis.service.UserService;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+import de.tum.in.www1.artemis.web.rest.util.HeaderUtil;
+import io.github.jhipster.web.util.ResponseUtil;
+
+/**
+ * REST controller for managing complaints.
+ */
+@RestController
+@RequestMapping("/api")
+public class ComplaintResponseResource {
+
+    private final Logger log = LoggerFactory.getLogger(SubmissionResource.class);
+
+    private static final String ENTITY_NAME = "complaintResponse";
+
+    private ComplaintRepository complaintRepository;
+
+    private ComplaintResponseRepository complaintResponseRepository;
+
+    private UserService userService;
+
+    public ComplaintResponseResource(ComplaintRepository complaintRepository, ComplaintResponseRepository complaintResponseRepository, UserService userService) {
+        this.complaintRepository = complaintRepository;
+        this.complaintResponseRepository = complaintResponseRepository;
+        this.userService = userService;
+    }
+
+    /**
+     * POST /complaint-responses: create a new complaint response
+     *
+     * @param complaintResponse the complaint response to create
+     * @return the ResponseEntity with status 201 (Created) and with body the new complaint response
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("/complaint-responses")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<ComplaintResponse> createComplaint(@RequestBody ComplaintResponse complaintResponse) throws URISyntaxException {
```

Comment:
please rename to createComplaintResponse

Reply:
Right, thanks for noticing!